generated/nidigital/session.py

```diff
@@ -2687,6 +2749,48 @@ def self_test(self):
             raise errors.SelfTestError(code, msg)
         return None
 
+    @ivi_synchronized
+    def write_source_waveform_site_unique(self, waveform_name, waveform_data):
+        '''write_source_waveform_site_unique
+
+        TBD
+
+        Args:
+            waveform_name (str):
+
+            waveform_data ({ site: data, site: data, ... }): Dictionary where each key is the site number and the value is array.array of unsigned int
+
+        '''
+        if len(waveform_data) == 0:
+            # Nothing to do
+            return
+
+        site_list = []
+        # We assume all the entries are the same length (we'll check later) to make the array the correct size
+        # Get an entry from the dictionary from https://stackoverflow.com/questions/30362391/how-do-you-find-the-first-key-in-a-dictionary
+        actual_samples_per_waveform = len(waveform_data[next(iter(waveform_data))])
+        data = array.array('L', [0] * (len(waveform_data) * actual_samples_per_waveform))
+        mv = memoryview(data)
+
+        i = 0
+        for site in waveform_data:
+            if len(waveform_data[site]) != actual_samples_per_waveform:
+                raise ValueError('Mismatched length of waveforms. All must be the same length.')
+            if waveform_data[site].typecode != 'L':
+                raise ValueError('Wrong array element type. Must be unsigned 32 bit int "l", was {}'.format(waveform_data[site].typecode))
+
+            site_list.append(str(site))
+
+            start = i * actual_samples_per_waveform
+            end = start + actual_samples_per_waveform
+            mv[start:end] = waveform_data[site]
+
+            i += 1
+
+        site_list_str = ','.join(site_list)
```

Comment:
We need to create a string of the form `"site0,site1"` before passing it down to C API.

Before 1.0, we need to figure out the right type to use for sites across the API. Tracked in #1111.

Reply:
Fixed